source/slang/slang-options.cpp

```diff
@@ -857,6 +858,26 @@ struct OptionsParser
                         return SLANG_FAIL;
                     }
                 }
+                else if (argStr == "-r")
+                {
+                    String moduleName;
+                    SLANG_RETURN_ON_FAIL(tryReadCommandLineArgument(sink, arg, &argCursor, argEnd, moduleName));
+
+                    // We need to deserialize and add the modules
+                    FileStream fileStream(moduleName, FileMode::Open, FileAccess::Read, FileShare::ReadWrite);
+
+                    List<RefPtr<IRModule>> irModules;
+                    if (SLANG_FAILED(IRSerialReader::readStreamModules(&fileStream, asInternal(session), requestImpl->getFrontEndReq()->getSourceManager(), irModules)))
+                    {
+                        sink->diagnose(SourceLoc(), Diagnostics::unableToReadModuleContainer, moduleName);
+                        return SLANG_FAIL;
+                    }
+
+                    // TODO(JS): May be better to have a ITypeComponent that encapsulates a collection of modules
```

Comment:
Typo in comment: `ITypeComponent` -> `IComponentType`.
Also, as discussed elsewhere, the logical way to build an `IComponentType` to encapsulate a collection of modules would be to wrap them each as a `Module` and then build a `CompositeComponentType`.

Reply:
This logic seems like it would need/want to move behind an `sp*()` API function so that there is a way to achieve the same result as `-r` via the API instead...